src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -230,8 +254,57 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
                 }
                 else
                 {
-                    StartWorkerProcesses(0, InitializeJobhostLanguageWorkerChannelAsync);
+                    // if _workerIndexing, initialize a single channel and let the rest start up the background
+                    if (_workerIndexing)
+                    {
+                        await InitializeJobhostLanguageWorkerChannelAsync();
+                        StartWorkerProcesses(1, InitializeJobhostLanguageWorkerChannelAsync);
+                    }
+                    else
+                    {
+                        StartWorkerProcesses(0, InitializeJobhostLanguageWorkerChannelAsync);
+                    }
+                }
+            }
+        }
+
+        // Gets metadata from worker
+        public async Task<IEnumerable<FunctionMetadata>> GetWorkerMetadata()
+        {
+            var channels = _channels.ToList();
+            if (channels.Count > 0)
+            {
+                return await channels[0].WorkerGetFunctionMetadata();
```

Comment:
Can use channels.FirstOrDefault()?.WorkerGetFunctionMetadata()

Reply:
Good call, I'll change this to channels.First() because we already check to see if channels has at least one element in it